examples/interactionsDemo.html

```diff
@@ -0,0 +1,526 @@
+                 *  does not care about permissions to view interactions.
+                 */
+                $('#submitConversationIDButton').on('click', function() {
+                    let conversationId = $('#conversationID').val().trim();
+                    clientApp.coreUi.showInteractionDetails(conversationId);
+                });
+
+                // Authenticate with PureCloud
+                let authenticatingEl = document.querySelector('.authenticating');
+                setHidden(authenticatingEl, false);
+
+                let authenticated = false;
+                let canViewInteraction = false;
+                let canViewWrapUpCodes = false;
+
+                client.loginImplicitGrant(clientId, redirectUrl, { state: ('pcEnvironment=' + pcEnvironment) })
+                    .then(function () {
+                        authenticated = true;
+                        return new platformClient.UsersApi().getUsersMe({ 'expand': ['authorization'] });
+                    })
+                    .then(function (profileData) {
+                        this.profileData = profileData;
+                        $('#name').text('Name: ' + profileData.name);
+
+                        // Check if the current user will be able to even view the interaction before making the request
+                        let permissions = profileData.authorization.permissions;
+                        if (checkPermission(permissions, 'analytics:conversationDetail:view') &&
+                            (checkPermission(permissions, 'conversation:communication:view') ||
+                             checkPermission(permissions, 'quality:evaluation:add') ||
+                             checkPermission(permissions, 'quality:calibration:view') ||
+                             checkPermission(permissions, 'quality:evaluation:editScore'))) {
+                                canViewInteraction = true;
+                                $('#role').text('Role: Supervisor');
+                        } else {
+                                $('#role').text('Role: Agent');
+                        }
+
+                        if (checkPermission(permissions, 'routing:wrapupCode:view')) {
+                            canViewWrapUpCodes = true;
+                        }
+
+                           setHidden(authenticatingEl, true);
+                           makeSDKRequest();
+                    })
+                    .catch(function (err) {
+                        setHidden(authenticatingEl, true);
+                        if (!authenticated) {
+                            setErrorState('Failed to Authenticate with PureCloud');
+                        }
+                    });
+
+                /**
+                 * Makes the request to the API to get the current users interactions for
+                 * the past 30 days.
+                 */
+                function makeSDKRequest() {
```

Comment:
maybe pass the userId in here in this function so you don't have to rely on this?  You can close over the userId when building the reload click handler too so you won't have to deal with this at all.